### PR TITLE
Add support for Jenkins DSL pipeline

### PR DESCRIPTION
**Pull request by nfalco79**
Add support for Jenkins DSL pipeline.
Fix some issues in check validation.
Add support for i18n for messages in the main class.
Add some missing javadoc and unit test

### CODE REVIEW

**Comment by nfalco79** [comment]
@ctron the name for the upload command is _pdrone_. If you prefer a different name please advise me.

File: package-drone/src/main/resources/de/dentrassi/pm/jenkins/Messages.properties [hunk 1]
```diff
@@ -0,0 +1,17 @@
+# Copyright (c) 2015 IBH SYSTEMS GmbH.
```
Comment (ctron) [inline package-drone/src/main/resources/de/dentrassi/pm/jenkins/Messages.properties]: This file seems all new. So the first line would be: `Copyright (c) 2017 Nikolas Falco.`

Comment (ctron) [reply package-drone/src/main/resources/de/dentrassi/pm/jenkins/Messages.properties]: Same for some of the other file.

**Review by ctron — DISMISSED** [review]
This looks great. Thanks! Only a small remark about the copyright headers. Its your work 😉

**Comment by nfalco79** [comment]
I've change the copyright as you suggest. I post a screen of the pipeline snippet generator page
schermata 2017-06-07 alle 00 01 14

I came back about the removal of the `Serializable` interface from `DroneRecorder`. The reason is that the `UploadFiles` is not a static class and in the `invoke` method it uses a bunch of fields of the DroneRecorder instance that if not serialisable breaks the builds on slave Jenkins nodes.

It's not clear the reason to force the build step to be Serialisable just only for a bunch of fields. Why `serverUrl` yes, and instead `excludes` pattern no?
Maybe next PR proposes breaks this linkage.

**Review by ctron — APPROVED** [review]

**Comment by ctron** [comment]
I really don't know about the serialization stuff I Jenkins. So I can't even really guess. It think making a proper test setup would help. But I really don't have the some for this. Sorry.

**Comment by ctron** [comment]
Let me know when you think it is time for another release.

**Comment by nfalco79** [comment]
To be ready for release in my opinion we need
* Collects the artifacts uploaded to fill in the correct way the BuildData
* Fix the UploadFile class because MasterToSlave callable are serialised from master to slave and only the return value of `invoke` is serialised back (not the entire state of class) so we need to change the way to get flags back and artifacts
* Remove any dependencies from UploadFile of the DroneRecorder so avoid serialisation issues
* Render DroneRecorder more confortable for testing.
* adds a lot of unit test

**Comment by ctron** [comment]
@nfalco79 If you think a new release is a good idea, just raise an issue and I will do it.

**Comment by nfalco79** [comment]
@ctron we are ready to release, I sent to you an email but maybe that one in the pom.xml is not valid.

**Comment by ctron** [comment]
It is and I think I did reply. I am currently on my way back from EclipseCon France and will do this early next week. Hope this is still ok for you.

**Pull request merged**